### PR TITLE
chore: update node.js version handling for CLI

### PR DESCRIPTION
**Pull request by fossamagna**
#### Description of changes

- Lambda resource handlers will use 16.x by default
- Lambda functions and triggers will use 16.x by default

#### Issue #, if available

fix: https://github.com/aws-amplify/amplify-cli/issues/11252

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-analytics/src/provider-utils/awscloudformation/cloudformation-templates/pinpoint-cloudformation-template.json [hunk 1]
```diff
@@ -121,13 +119,9 @@
             {
               "Effect": "Allow",
               "Principal": {
-                "Service": [
-                  "lambda.amazonaws.com"
-                ]
+                "Service": ["lambda.amazonaws.com"]
```
Comment (danielleadams) [inline packages/amplify-category-analytics/src/provider-utils/awscloudformation/cloudformation-templates/pinpoint-cloudformation-template.json:122]: There seem to be a lot of unrelated formatting changes. Do you have a linter that is turned on? Is there any chance that you can separate the linter changes from the upgrade changes? This makes the pull request more difficult to review with all the additional changes.

Comment (fossamagna) [reply packages/amplify-category-analytics/src/provider-utils/awscloudformation/cloudformation-templates/pinpoint-cloudformation-template.json]: @danielleadams Thanks for the suggestion. I have turned off the linter. I have committed and pushed only changes that update the nodejs version.

Comment (danielleadams) [reply packages/amplify-category-analytics/src/provider-utils/awscloudformation/cloudformation-templates/pinpoint-cloudformation-template.json]: great - thank you.

**Comment by danielleadams** [comment]
@fossamagna I started an integration test run for CLI. Not sure why its not updating the GitHub PR, but I will keep an eye on it to see that it passes.

**Comment by danielleadams** [comment]
@fossamagna running your changes through the test suite.

**Comment by GeorgeBellTMH** [comment]
It would be good if before pushing we checked to ensure that both the current and updated node versions are supported by AWS. If the current one isn't supported we may not be able to roll back, if the future one isn't then it shouldn't push. In both cases there should be an override with some warning about how to manually fix if a rollback gets stuck (ie. you have to go in and edit the lambda directly).

**Comment by danielleadams** [comment]
@fossamagna The tests seem to have stalled at the very end, so I'm going to restart them, but I've marked this as semver-major, so they will go out in v11 (don't have a date for that yet).

**Comment by danielleadams** [comment]
> It would be good if before pushing we checked to ensure that both the current and updated node versions are supported by AWS. 

@GeorgeBellTMH Both Node.js 14 and 16 are supported by AWS Lambda: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html.

**Comment by GeorgeBellTMH** [comment]
They won't be forever...

**Comment by sobolk** [comment]
@GeorgeBellTMH 
since this change warrants major version bump. could you please target this PR to `cdkv2` branch where we're staging V11?

**Comment by sobolk** [comment]
@fossamagna 
since this change warrants major version bump. could you please target this PR to `cdkv2` branch where we're staging V11?

**Comment by fossamagna** [comment]
@danielleadams @sobolk Thanks, I changed PR target to `cdkv2` branch. I look forward to V11.

**Review by sobolk — APPROVED** [review]
